tests/trinity/integration/test_boot.py

```diff
@@ -89,6 +100,33 @@ def async_process_runner():
     })
 
 
+@pytest.mark.parametrize(
+    'command',
+    (
+        ['trinity'],
+        # ['trinity', '--light', '--ropsten'],
+    )
+)
+@pytest.mark.asyncio
+async def test_web3(event_loop, command, async_process_runner, second_async_process_runner):
+    await async_process_runner.run(command, timeout_sec=30)
+    assert await contains_all(async_process_runner.stderr, {
+        "Started DB server process",
+        "Started networking process",
+        "IPC started at",
+    })
+    await second_async_process_runner.run(['trinity', 'attach'], timeout_sec=30)
+    assert await contains_all(second_async_process_runner.stdout, {
+        "An instance of Web3",
+        "In [1]:",
+    })
+
+    stdout, stderr = await second_async_process_runner.proc.communicate(b'w3\n')
```

Comment:
@carver you like to tinker with puzzles, don't you? See, I've tried to spin up two trinity instances here and I get the point where the second instance that is running `trinity attach` should be waiting for input (we wait for `In [1]:` to be present in the `stdout`) But from there I haven't had any luck to interact with the shell. Not trying to offload debugging to you but it might be one of these things you know a simple answer to :)

Reply:
Hm, nothing obvious comes to mind. You've tried using the default console instead of ipython?


But... it does remind me that it would be nice to be able to pass commands over stdin sometimes, like:

```
echo "w3.eth.blockNumber" | trinity attach
```

We could add that feature, which would be easier to test. But that doesn't really help test the console.

Reply:
I tried this again today using the default python shell. I pushed another commit on top that does that but it behaves the same.

The test gets to the point where the attach command is waiting for stdin but that seems to never reach it. One can observe that by running the test with `pytest -s tests/trinity/integration/test_trinity_cli.py -k test_web3`

```sh
tests/trinity/integration/test_trinity_cli.py b'    INFO  08-06 15:33:01        main  \n'
b'      ______     _       _ __       \n'
b'     /_  __/____(_)___  (_) /___  __\n'
b'      / / / ___/ / __ \\/ / __/ / / /\n'
b'     / / / /  / / / / / / /_/ /_/ / \n'
b'    /_/ /_/  /_/_/ /_/_/\\__/\\__, /  \n'
b'                           /____/   \n'
b'    INFO  08-06 15:33:01        main  Trinity/eth-0.2.0a29/linux/cpython3.6.5\n'
b'    INFO  08-06 15:33:01        main  Trinity DEBUG log file is created at /tmp/pytest-of-cburgdorf/pytest-357/test_web3_command0_0/trinity/mainnet/logs/trinity.log\n'
b'    INFO  08-06 15:33:01        main  Started DB server process (pid=15355)\n'
b'    INFO  08-06 15:33:02        main  Started networking process (pid=15361)\n'
b'    INFO  08-06 15:33:03      server  Running server...\n'
b'    INFO  08-06 15:33:03         nat  Setting up NAT portmap...\n'
b'    INFO  08-06 15:33:03         ipc  IPC started at: /tmp/pytest-of-cburgdorf/pytest-357/test_web3_command0_0/trinity/mainnet/jsonrpc.ipc\n'
b'    INFO  08-06 15:33:04        main  \n'
b'      ______     _       _ __       \n'
b'     /_  __/____(_)___  (_) /___  __\n'
b'      / / / ___/ / __ \\/ / __/ / / /\n'
b'     / / / /  / / / / / / /_/ /_/ / \n'
b'    /_/ /_/  /_/_/ /_/_/\\__/\\__, /  \n'
b'                           /____/   \n'
b'    INFO  08-06 15:33:04        main  Trinity/eth-0.2.0a29/linux/cpython3.6.5\n'
b'    INFO  08-06 15:33:04        main  Trinity DEBUG log file is created at /tmp/pytest-of-cburgdorf/pytest-357/test_web3_command0_0/trinity/mainnet/logs/trinity.log\n'
b'Trinity Console\n'
b'---------------\n'
b'An instance of Web3 connected to the running chain is available as the `w3` variable\n'
b'\n'
b''
F
```

:cry:

Reply:
Lets shelve this for now and we can try to stab at it again later.

Reply:
Yep, I already gave this a lower priority in my head. It's just something that bugs me but I'll probably need to cut it down to a smaller repro that I can put into a SO post and maybe throw a bounty add it.